src/services/action/create-action-task.ts

```diff
@@ -180,13 +36,18 @@ export class CreateActionTask extends BaseActionTask<Action> {
 
     log.debug('create action');
 
-    const { request, reply } = this.input;
+    const { request, reply, handler: buildActionHandler } = this.input;
 
     // create action only on successful requests
     if (reply.statusCode >= 200 && reply.statusCode < 300) {
-      const getItemFromDb = (id) => this.itemService.get(id, handler);
-      const actions = await buildActionsFromRequest(request, getItemFromDb, this.hosts, log);
-
+      // new version with the handler
+      const actionInput: ActionHandlerInput = {
+        request,
+        reply,
+        dbHandler: handler,
+        log,
+      };
+      const actions = await buildActionHandler(actionInput);
```

Comment:
`buildActionHandler` makes me think that you are building the handler (and not the actions). Maybe you can rename it `generateActions`?

Reply:
I didn't see a check to see if the handler was defined (since it is potentially optional). Which case would it be undefined for? Maybe have a test for an undefined handler.

Reply:
It should alway be defined, i removed the optional.